tests/func/test_ls.py

```diff
@@ -0,0 +1,105 @@
+import shutil
+import os
+
+from dvc.repo import Repo
+
+FS_STRUCTURE = {
+    "file1": "",
+    "dir1/file2": "",
+    "dir1/file3": "",
+    "dir2": {},
+    ".gitignore": "",
+}
+
+DVC_STRUCTURE = {"dvc_file1": "content", "dir1/dvc_file2": "content"}
+
+
+def test_ls_repo(tmp_dir, dvc, scm):
+    tmp_dir.scm_gen(FS_STRUCTURE, commit="init")
+    tmp_dir.dvc_gen(DVC_STRUCTURE, commit="dvc")
+
+    files = Repo.ls(tmp_dir)
+    assert set(files) == set(
+        ["file1", "dir1", "dir2", ".gitignore", "dvc_file1", "dvc_file1.dvc"]
+    )
+
+
+def test_ls_repo_recursive(tmp_dir, dvc, scm):
+    tmp_dir.scm_gen(FS_STRUCTURE, commit="init")
+    tmp_dir.dvc_gen(DVC_STRUCTURE, commit="dvc")
+
+    files = Repo.ls(tmp_dir, recursive=True)
+    assert set(files) == set(
+        [
+            "file1",
+            "dir1",
+            "dir1/file2",
```

Comment:
This is the reason why windows tests are failing
```suggestion
            os.path.join("dir1", "file2"),
```
and so on down below